cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +105,300 @@ def unset_default_org(self):
+        present as the defaults for their given types, and write these
+        out to a DEFAULT_SERVICES.json file in the directory. This occurs
+        once before .service files are migrated to the appropriate
+        services/ sub-directory, so we set the default to the alias
+        that will be assigned during migration.
+
+        @param dir_path: the directory to look through and write
+        the DEFAULT_SERVICES.json file in.
+        """
+        if dir_path is None:
+            return  # pragma: no cover
+
+        dir_path = Path(dir_path)
+        default_services = {}
+        for item in dir_path.iterdir():
+            if item.suffix == ".service":
+                service_type = item.name.replace(".service", "")
+                scope = "global" if item.parent.name == ".cumulusci" else "project"
+                default_services[service_type] = f"{service_type}__{scope}"
+
+        with open(dir_path / DEFAULT_SERVICES_FILENAME, "w") as f:
+            f.write(json.dumps(default_services))
+
+    def _create_services_dir_structure(self) -> None:
+        """
+        Ensure the 'services' directory structure exists.
+        The services dir has the following structure and lives
+        in the global_config_dir:
+
+        services
+        |-- github
+        |   |-- alias1.service
+        |   |-- alias2.service
+        |   |-- ...
+        |-- devhub
+        |   |-- alias1.service
+        |   |-- alias2.service
+        |   |-- ...
+        .
+
+        This also has the advantage that when a new service type
+        is added to cumulusci.yml a new directory for that service type
+        will be created.
+        """
+        services_dir_path = Path(f"{self.global_config_dir}/services")
+        if not Path.is_dir(services_dir_path):
+            Path.mkdir(services_dir_path)
+
+        configured_service_types = self.project_config.config["services"].keys()
+        for service_type in configured_service_types:
+            service_type_dir_path = Path(services_dir_path / service_type)
+            if not Path.is_dir(service_type_dir_path):
+                Path.mkdir(service_type_dir_path)
+
+    def _migrate_services(self):
+        """Migrate .service files from the global_config_dir and
+        any project local directories."""
+        self._migrate_services_from_dir(self.global_config_dir)
+        for local_proj_dir in self._iter_local_project_dirs():
+            self._migrate_services_from_dir(local_proj_dir)
+
+    def _migrate_services_from_dir(self, dir_path: str) -> None:
+        """Migrate all .service files from the given directory to
+        the appropriate service sub-directory and apply the default
+        alias. This is intended to be run against either the
+        global_config_dir, or a local project directory.
+
+        Default aliases are in the form `service_type__scope`.
+        Scope is either 'project' or 'global' depending
+        on where the .service file is located."""
```

Comment:
I take it you decided not to try to support multiple projects with project-specific services by naming them based on the project name?

Reply:
This was something I just forgot to incorporate. I'll go back and name them based on project name.